docs/ECLLanguageReference/ECLR_mods/Basics-Constants.xml

```diff
@@ -152,4 +152,100 @@ MyReal1 := 10.0;   // value of MyReal1 is the REAL value 10.0
+    creating conditional code.</para>
+
+    <para><informaltable colsep="1" frame="all" rowsep="1">
+        <tgroup cols="2">
+          <colspec colwidth="150pt" />
+
+          <colspec />
+
+          <tbody>
+            <row>
+              <entry><emphasis>_ECL_VERSION_<indexterm>
+                  <primary>_ECL_VERSION_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A STRING containing the value of the platform version.
+              For example, '6.4.0'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_VERSION_MAJOR_ <indexterm>
+                  <primary>_ECL_VERSION_MAJOR_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>An INTEGER containing the value of the major portion of
+              the platform version. For example, '6'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_VERSION_MINOR_<indexterm>
+                  <primary>_ECL_VERSION_MINOR_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>An INTEGER containing the value of the minor portion of
+              the platform version. For example, '4'</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_ECL_LEGACY_MODE_<indexterm>
+                  <primary>_ECL_LEGACY_MODE_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A BOOLEAN value indicating if it is being compiled with
+              legacy IMPORT semantics.</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_OS_<indexterm>
+                  <primary>_OS_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A STRING indicating the operating system to which it is
+              being compiled. (Available values are: 'windows' or
+              'linux')</entry>
+            </row>
+
+            <row>
+              <entry><emphasis>_STAND_ALONE_<indexterm>
+                  <primary>_STAND_ALONE_</primary>
+                </indexterm></emphasis></entry>
+
+              <entry>A BOOLEAN value indicating if it is being compiled to a
+              stand-alone executable.</entry>
+            </row>
+          </tbody>
+        </tgroup>
+      </informaltable>Example:</para>
+
+    <para><programlisting>IMPORT STD;
+  STRING14 fGetDateTimeString() :=
+  #IF(__ECL_VERSION__ &gt;= '5.2')
+    STD.Date.SecondsToString(STD.Date.CurrentSeconds(true), '%Y%m%d%H%M%S');
+  #ELSE
+    FUNCTION
+      string14 fGetDimeTime():= // 14 characters returned
+      BEGINC++
+      #option action
+      struct tm localt;         // localtime in "tm" structure
+      time_t timeinsecs;        // variable to store time in secs
+      time(&amp;timeinsecs);  
+      localtime_r(&amp;timeinsecs,&amp;localt);
+      strftime(__result, 15, "%Y%m%d%H%M%S", &amp;localt); // Formats the localtime to YYYYMMDDhhmmss
```

Comment:
Bad code - this corrupts memory.  It needs to write to a temporary, and then copy 14 bytes from it.